dwave/plugins/sklearn/transformers.py

```diff
@@ -0,0 +1,293 @@
+        y: npt.ArrayLike,
+        *,
+        alpha: float,
+        num_features: int,
+        strict: bool = True,
+    ) -> dimod.ConstrainedQuadraticModel:
+        """Build a constrained quadratic model for feature selection.
+
+        This method is based on maximizing influence and independence as
+        measured by correlation [Milne et al.]_.
+
+        Args:
+            X:
+                Feature vectors formatted as a numerical 2D array-like.
+            y:
+                Class labels formatted as a numerical 1D array-like.
+            alpha:
+                Hyperparameter between 0 and 1 that controls the relative weight of
+                the relevance and redundancy terms.  `alpha=1` places all weight on
+                relevance and selects all features, whereas `alpha=0` places all
+                weight on redundancy and selects no features.
+            num_features:
+                The number of features to select.
+            strict:
+                If ``False`` the constraint on the number of selected features
+                is ``<=`` rather than ``==``.
+
+        Returns:
+            A constrained quadratic model.
+
+        .. [Milne et al.] Milne, Andrew, Maxwell Rounds, and Phil Goddard. 2017. "Optimal Feature
+            Selection in Credit Scoring and Classification Using a Quantum Annealer."
+            1QBit; White Paper.
+            https://1qbit.com/whitepaper/optimal-feature-selection-in-credit-scoring-classification-using-quantum-annealer
+        """
+
+        X = np.atleast_2d(np.asarray(X))
+        y = np.asarray(y)
+
+        if X.ndim != 2:
+            raise ValueError("X must be a 2-dimensional array-like")
+
+        if y.ndim != 1:
+            raise ValueError("y must be a 1-dimensional array-like")
+
+        if not 0 <= alpha <= 1:
+            raise ValueError(f"alpha must be between 0 and 1, given {alpha}")
+
+        if num_features <= 0:
+            raise ValueError(f"num_features must be a positive integer, given {num_features}")
+
+        cqm = dimod.ConstrainedQuadraticModel()
+        cqm.add_variables(dimod.BINARY, X.shape[1])
+
+        # add the k-hot constraint
+        cqm.add_constraint(((v, 1) for v in cqm.variables), '==' if strict else '<=', num_features)
+
+        with tempfile.TemporaryFile() as fX, tempfile.TemporaryFile() as fout:
+            # we make a copy of X because we'll be modifying it in-place within
+            # some of the functions
+            X_copy = np.memmap(fX, X.dtype, mode="w+", shape=(X.shape[0], X.shape[1] + 1))
+            X_copy[:, :-1] = X
+            X_copy[:, -1] = y
+
+            # make the matrix that will hold the correlations
+            correlations = np.memmap(
+                fout,
+                dtype=np.result_type(X, y),
+                mode="w+",
+                shape=(X_copy.shape[1], X_copy.shape[1]),
+                )
+
+            # main calculation. It modifies X in-place
```

Comment:
```suggestion
            # main calculation. It modifies X_copy in-place
```

It modifies the ``x`` input of ``corrcoef(x: npt.ArrayLike, *, ...`` but here it does not modify ``X``